### PR TITLE
External CI: Dockerless + latest source for rocprofiler and rocm_bandwidth_test

### PR DESCRIPTION
**Pull request by danielsu-amd**
Both rocprofiler and rocm_bandwidth_test now successfully build on their latest source.

rocprofiler:
https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=1477&view=logs&j=d5e2a0d6-4bff-5e6c-19be-0282ec1028db
rocm_bandwidth_test:
https://dev.azure.com/ROCm-CI/ROCm-CI/_build/results?buildId=1483&view=logs&j=4cbbbbe6-e3e9-50e2-34c4-5262ef5a7519

### CODE REVIEW

File: .azuredevops/components/rocprofiler.yml [hunk 1]
```diff
@@ -46,11 +62,68 @@ jobs:
   - template: ${{ variables.CI_TEMPLATE_PATH }}/steps/checkout.yml
     parameters:
       checkoutRepo: ${{ parameters.checkoutRepo }}
+# Manually download rocm-core and aqlprofile, hard-coded 6.1.0 versions
```
Comment (danielsu-amd) [inline .azuredevops/components/rocprofiler.yml:65]: Should I use a variable for the rocm-core and aqlprofile versions? I haven't yet tried using other versions of these dependencies.

File: .azuredevops/components/rocprofiler.yml [hunk 2]
```diff
@@ -46,11 +62,68 @@ jobs:
...
+        mkdir rocm-core
+        dpkg-deb -R rocm-core_6.1.0.60100-82~22.04_amd64.deb rocm-core
+      workingDirectory: '$(Pipeline.Workspace)'
+  - task: Bash@3
+    displayName: 'Move rocm-core'
+    inputs:
+      targetType: inline
+      script: |
+        mkdir -p $(Agent.BuildDirectory)/rocm
+        cp -R rocm-core/opt/rocm-6.1.0/* $(Agent.BuildDirectory)/rocm
```
Comment (danielsu-amd) [inline .azuredevops/components/rocprofiler.yml:86]: Is this method of extracting and copying the files into the right /rocm directory sensible?

File: .azuredevops/components/rocprofiler.yml (hunk 1 above)
Comment (amd-jmacaran) [reply .azuredevops/components/rocprofiler.yml:65]: This is fine for now to get live builds enabled. We have to refactor how aqlprofile gets incorporated in the future anyway. As for rocm-core, you can merge this solution and then next investigate what that debian package actually installs and see if we can get one of our existing pipelines to replicate producing those files.

File: .azuredevops/components/rocprofiler.yml (hunk 2 above)
Comment (amd-jmacaran) [reply .azuredevops/components/rocprofiler.yml:86]: This is fine for now.

**Review by amd-jmacaran — APPROVED** [review]

**Pull request merged**